src/core/lib/event_engine/windows/resolved_address.cc

```diff
@@ -0,0 +1,97 @@
+// Copyright 2022 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include <grpc/support/port_platform.h>
+
+#ifdef GPR_WINDOWS
+
+#include <errno.h>
+#include <inttypes.h>
+#include <winsock2.h>
+
+#include "absl/strings/str_format.h"
+
+#include "src/core/lib/event_engine/windows/resolved_address.h"
+#include "src/core/lib/gprpp/host_port.h"
+#include "src/core/lib/gprpp/status_helper.h"
+#include "src/core/lib/uri/uri_parser.h"
+
+namespace grpc_event_engine {
+namespace experimental {
+
+namespace {
+absl::StatusOr<std::string> GetScheme(
+    const EventEngine::ResolvedAddress& resolved_address) {
+  if (resolved_address.address()->sa_family == AF_INET) return "ipv4";
+  if (resolved_address.address()->sa_family == AF_INET6) return "ipv6";
+  return absl::InvalidArgumentError(absl::StrFormat(
+      "Unknown scheme: %d", resolved_address.address()->sa_family));
+}
+}  // namespace
+
+absl::StatusOr<std::string> ResolvedAddressToURI(
+    const EventEngine::ResolvedAddress& resolved_address) {
+  if (resolved_address.size() == 0) {
+    return absl::InvalidArgumentError("Empty address");
+  }
+  auto scheme = GetScheme(resolved_address);
+  GRPC_RETURN_IF_ERROR(scheme.status());
+  auto path = ResolvedAddressToString(resolved_address);
+  GRPC_RETURN_IF_ERROR(path.status());
+  absl::StatusOr<grpc_core::URI> uri =
+      grpc_core::URI::Create(*scheme, /*authority=*/"", std::move(path.value()),
+                             /*query_parameter_pairs=*/{}, /*fragment=*/"");
+  if (!uri.ok()) return uri.status();
+  return uri->ToString();
+}
+
+// TODO(hork): implement normalization
+absl::StatusOr<std::string> ResolvedAddressToString(
```

Comment:
I count two implementations of this in the code base now, making this a third.

Given this is relatively complex and bug-prone code to write & maintain, I'd like us to have a plan to get down to one. I'd strongly prefer that we never exceed two -- can this implementation be merged with another to achieve that?

Reply:
This is done, but because of it, I think we'll need a cherrypick.

Reply:
Correction: this does not require a cherrypick, I did a dry run and found no uses (which makes sense)